lib/gcs_signer.ex

```diff
@@ -66,18 +73,30 @@ defmodule GcsSigner do
 
     string
     |> :public_key.sign(:sha256, private_key)
-    |> Base.encode64
+    |> Base.encode64()
   end
 
   # Derive private key from Google Cloud Service Account
   # Idea for this code is mostly from this GitHub issue:
   # https://github.com/potatosalad/erlang-jose/issues/13#issuecomment-160718744
   defp get_private_key(client) do
     client.private_key
-    |> :public_key.pem_decode
+    |> :public_key.pem_decode()
     |> (fn [x] -> x end).()
-    |> :public_key.pem_entry_decode
-    |> elem(3) # grab privateKey from the record tuple
-    |> (fn pk -> :public_key.der_decode(:RSAPrivateKey, pk) end).()
+    |> :public_key.pem_entry_decode()
+    |> normalize_private_key()
+  end
+
+  defp normalize_private_key(private_key) do
+    if @otp_greater_21? do
+      # From OTP 21, GCS keys are correctly decoded and do not need any
+      # extra treatment
+      private_key
+    else
+      private_key
+      # grab privateKey from the record tuple
```

Comment:
I'd like to suggest moving this comment to above L96, since comment is in the middle of the pipe methods which makes `private_key` and subsequent method disconnected.